cron/uptycs_alertpull.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+import sys
+from datetime import datetime, timedelta, tzinfo
+
+try:
+    from datetime import timezone
+
+    utc = timezone.utc
+except ImportError:
+    # Hi there python2 user
+    class UTC(tzinfo):
+        def utcoffset(self, dt):
+            return timedelta(0)
+
+        def tzname(self, dt):
+            return "UTC"
+
+        def dst(self, dt):
+            return timedelta(0)
+
+    utc = UTC()
+from configlib import getConfig, OptionParser
+import json
+import mozdef_client as mozdef
+import pickle
+import datetime
+import pdb
+import json
+import jwt
+import datetime
+import requests
+import urllib3
+import click
+import os
+import sys
+
+
+class UptycsFilter(object):
+    def __init__(self):
+        utc_now = datetime.datetime.utcnow()
+        self.start_time = utc_now - datetime.timedelta(days=2)
+        self.end_time = utc_now
+        self.sort_style = 'alertTime:desc'
+        self.limit = 100
+        self.offset = 0
+
+    def filter_string(self):
+        return json.dumps({
+            "alertTime": {
+                "between": [
+                    self.start_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+                    self.end_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+                ]
+            },
+            "status": {"in": ["open", "assigned"]}
+        }, sort_keys=False, separators=(',', ':'))
+
+
+class UptycsClient(object):
+    def __init__(self, path):
+        self.api_json = json.load(open(path))
+        self.url = 'https://' + self.api_json['domain'] + '.uptycs.io'
+        self.verify_ssl = True
+        self.customer_id = self.api_json['customerId']
+        self.key = self.api_json['key']
+        self.secret = self.api_json['secret']
+
+    def auth_headers(self):
+        headers = {}
+        utcnow = datetime.datetime.utcnow()
+        date = utcnow.strftime("%a, %d %b %Y %H:%M:%S GMT")
+        authVar = jwt.encode({'iss': self.key}, self.secret, algorithm='HS256')
+        authorization = "Bearer %s" % (authVar.decode('utf-8'))
+        headers['date'] = date
+        headers['Authorization'] = authorization
+        return headers
+
+    def get(self, path, params={}):
+        final_url = ("%s/public/api/customers/%s%s" %
+                     (self.url, self.customer_id, path))
+        try:
+            response = requests.get(url=final_url,
+                                    headers=self.auth_headers(),
+                                    verify=self.verify_ssl)
+            if (response and response.status_code in
+                    [requests.codes.ok, requests.codes.bad]):
+                return response.json()
+        except requests.exceptions.RequestException as e:
+            print("ERROR: {}".format(e))
```

Comment:
I presume we'll want better error/exception handling here to reset and preserve our statefile.